src/mjolnir/ingest_transit.cc

```diff
@@ -0,0 +1,778 @@
+    const auto& tile_stopId = feed_stop.id;
+    const auto& feed = stopFeeds(feed_stop);
+    auto* node = tile.add_nodes();
+    gtfs::Stop tile_stop = *(feed.get_stop(tile_stopId));
+    node->set_lon(tile_stop.stop_lon);
+    node->set_lat(tile_stop.stop_lat);
+    node->set_type((int)tile_stop.location_type);
+    node->set_graphid(node_id);
+    node_id++;
+    // node->set_prev_graphid(); what is prev?
+    node->set_name(tile_stop.stop_name);
+    // node->set_osm_way_id(); where is this?
+    node->set_timezone(tile_stop.stop_timezone);
+    // 0 is No Info ; 1 is True ; 2 is False
+    bool wheelchair_accessible = (tile_stop.wheelchair_boarding == "1");
+    node->set_wheelchair_boarding(wheelchair_accessible);
+    // node->set_generated(feed_stop.generated); ?
+    node->set_onestop_id(tile_stop.stop_id);
+    stop_graphIds[tile_stopId] = node_id;
+  }
+  return stop_graphIds;
+}
+
+// read feed data per stop, given shape
+float add_stop_pair_shapes(const gtfs::Stop& stop_connect,
+                           const gtfs::Shape& trip_shape,
+                           const gtfs::StopTime& pointStopTime) {
+  // check which segment would belong to which tile
+  if (pointStopTime.shape_dist_traveled > 0) {
+    return pointStopTime.shape_dist_traveled;
+  }
+  float dist_traveled = 0;
+  float min_sq_distance = INFINITY;
+  PointLL stopPoint = PointLL(stop_connect.stop_lon, stop_connect.stop_lat);
+  projector_t project(stopPoint);
+  for (int segment = 0; segment < (int)trip_shape.size() - 1; segment++) {
+    auto currOrigin = trip_shape[segment];
+    auto currDest = trip_shape[segment + 1];
+    PointLL originPoint = PointLL(currOrigin.shape_pt_lon, currOrigin.shape_pt_lat);
+    PointLL destPoint = PointLL(currDest.shape_pt_lon, currDest.shape_pt_lat);
+
+    PointLL minPoint = project(originPoint, destPoint);
+    float sq_distance = project.approx.DistanceSquared(minPoint);
+    if (sq_distance < min_sq_distance) {
+      min_sq_distance = dist_traveled + originPoint.Distance(minPoint);
+    }
+    dist_traveled += originPoint.Distance(destPoint);
+  }
+  return dist_traveled;
+}
+
+struct unique_transit_t {
+  std::mutex lock;
+  std::unordered_map<std::string, size_t> trips;
+  std::unordered_map<std::string, size_t> block_ids;
+  std::unordered_map<std::string, size_t> lines;
+};
+
+// return dangling stop_pairs, write stop data from feed
+bool write_stop_pairs(Transit& tile,
+                      const tileTransitInfo& tile_info,
+                      std::unordered_map<gtfs::Id, GraphId> stop_graphIds,
+                      unique_transit_t& uniques) {
+
+  const auto& tile_tripIds = tile_info.tile_trips;
+  feedCache tripFeeds;
+  bool dangles = false;
+
+  for (const feedObject& feed_trip : tile_tripIds) {
+    const auto& tile_tripId = feed_trip.id;
+    const std::string currFeedPath = feed_trip.feed;
+    const auto& feed = tripFeeds(feed_trip);
+
+    gtfs::Trip currTrip = *(feed.get_trip(tile_tripId));
```

Comment:
same here: can't be an invalid id?

Reply:
I won't leave this comment with all occurrences but there's a lot more. in case you thought about it while developing and decided it's fine, never mind:)

Reply:
I'll be doing the same as above -> if there exists no id for the object (whether it be route, trip, etc...), I will filter it out so all occurrences of search by id have a valid result.

Reply:
i forget if this one is referenced from another item  (like route for example) or if we got it by looping over trips. if we got it by looping over the thing directly we dont need to check if its not there now but if its derived from another thing (ie joined from another table) then yeah we s hould check it.  any place that you skip something you should definitely put a LOG_WARN about what feed and what thing and what id was getting skipped